src/objects.rs

```diff
@@ -691,6 +692,45 @@ impl GetObjectType for VehicleJourney {
     }
 }
 
+#[derive(Debug, Error)]
+pub enum StopTimeError {
+    #[error("duplicate stop_sequence '{duplicated_sequence}' for the trip '{vj_id}'")]
+    DuplicateStopSequence {
+        vj_id: String,
+        duplicated_sequence: u32,
+    },
+    #[error("incoherent stop times '{first_incorrect_sequence}' at time '{first_incorrect_time}' for the trip '{vj_id}'")]
+    IncoherentStopTimes {
+        vj_id: String,
+        first_incorrect_sequence: u32,
+        first_incorrect_time: Time,
+    },
+}
+
+impl VehicleJourney {
+    pub fn sort_and_check_stop_times(&mut self) -> Result<(), StopTimeError> {
+        self.stop_times.sort_unstable_by_key(|st| st.sequence);
+        for window in self.stop_times.windows(2) {
+            let curr_st = &window[0];
+            let next_st = &window[1];
+            if curr_st.sequence == next_st.sequence {
+                return Err(StopTimeError::DuplicateStopSequence {
+                    duplicated_sequence: curr_st.sequence,
+                    vj_id: self.id.clone(),
+                });
+            }
+            if curr_st.departure_time > next_st.arrival_time {
```

Comment:
```suggestion
            if curr_st.arrival_time > curr_st.departure_time || curr_st.departure_time > next_st.arrival_time {
```
Maybe better like this, because it can be wrong within a single stop point (arrival> departure)